tests/Signature/SignatureV4Test.php

```diff
@@ -277,6 +277,32 @@ public function testProvider()
     /**
      * @dataProvider testProvider
      */
+    public function testSignRequestUnsignedPayload($req, $sreq, $creq)
```

Comment:
A new provider would be one way to do it. You could also modify the existing testProvider to provide an additional parameter indicating if the request should be signed.